### DISCUSSION TITLE
Pyinstaller in python

### DISCUSSION THREAD

**Discussion by eli3z** [TomSchimansky/CustomTkinter#2196] [Q&A]
How use pyinstaller if my UI uses a customtkinter?

**Comment by Anonymous6598** [comment]
You have to use `add-folder` parameter. Basically, just add customtkinter folder path to you pyinstaller command.

**Reply by eli3z** [reply]
In my case my folder name is app_project and inside it is the py file I want to install which is located at C:\Users\eliez\Desktop\app_project, so my code will be pyinstaller --C:\Users\eliez\Desktop\app_project? And also I have a subfolders to that folder do I need to add that to the code?

**Reply by Anonymous6598** [reply]
You have to add a folder with customtkinter framework

**Reply by Anonymous6598** [reply]
`pyinstaller --noconfirm --onedir --console --add-data "C:/Program Files/Python312/Lib/site-packages/customtkinter;customtkinter/"  ""`

**Reply by Anonymous6598** [reply]
this command

**Reply by eli3z** [reply]
So basically like this

pyinstaller --noconfirm --onedir --console --add-data "C:/Program Files/Python312/Lib/site-packages/customtkinter;customtkinter/"  "C:/Users/eliez/Desktop/app_project/name.py" ?

Sorry I'm new to programming

**Reply by Anonymous6598** [reply]
Yes, like this.

**Reply by Anonymous6598** [reply]
What python version do you have?

**Reply by eli3z** [reply]
I am using python 3.10.8

**Reply by eli3z** [reply]
It's on Ubuntu 18.04 LTS

**Reply by Anonymous6598** [reply]
Than choose python310 folder and there find customtkinter framework

**Reply by Anonymous6598** [reply]
> So basically like this
> 
> pyinstaller --noconfirm --onedir --console --add-data "C:/Program Files/Python312/Lib/site-packages/customtkinter;customtkinter/"  "C:/Users/eliez/Desktop/app_project/name.py" ?
> 
> Sorry I'm new to programming 

pyinstaller --noconfirm --onedir --console --add-data "C:/Program Files/Python310/Lib/site-packages/customtkinter;customtkinter/"  "C:/Users/eliez/Desktop/app_project/name.py"

**Reply by Anonymous6598** [reply]
I hope you understand it. If it goes well, mark my comment as answer to your question.

**Reply by Anonymous6598** [reply]
Just for Ubuntu use bin folder. I use Windows. That's why there is C:/ drive

**Reply by Anonymous6598** [reply]
Better use auto-py-to-exe. It's easier.

**Comment by Akascape** [comment]
@eli3z Full detailed guide: https://www.akascape.com/coding/best-pyinstaller-guide

With video tutorial: https://youtu.be/6WApXONjGFA?si=W0HxQHz7GO7hozY2

**Reply by Anonymous6598** [reply]
@Akascape, that's good idea.

**Reply by Anonymous6598** [reply]
Your solution is good.
